README.md

```diff
@@ -323,13 +327,106 @@ and enable verbose logging:
 ## Usage
 
 Usage steps are [documented on MDN](https://developer.mozilla.org/en-US/docs/Mozilla/QA/Marionette/WebDriver),
-but the gist of it is this:
+but how you invoke geckodriver largely depends on your use case.
+
+### Selenium
+
+If you are using geckodriver through [Selenium](http://seleniumhq.org/),
+you must ensure that you have version 3.3.1 or greater.
+Because geckodriver implements the [W3C WebDriver standard](https://w3c.github.io/webdriver/webdriver-spec.html)
+and not the same Selenium wire protocol older drivers are using,
+you may experience incompatibilities and migration problems
+when making the switch from FirefoxDriver to geckodriver.
+
+Generally speaking, Selenium 3 enabled geckodriver
+as the default WebDriver implementation for Firefox.
+With the release of Firefox 47, FirefoxDriver had to be discontinued
+for its lack of support for the [new multi-processing architecture in Gecko](https://developer.mozilla.org/en-US/Firefox/Multiprocess_Firefox).
 
-	% geckodriver -b /usr/bin/firefox
+Selenium client bindings will pick up the _geckodriver_ binary executable
+from your [system’s `PATH` environmental variable](https://en.wikipedia.org/wiki/PATH_(variable))
+unless you override it by setting the `webdriver.gecko.driver` [Java VM system property](http://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html):
+
+```java
+System.setProperty("webdriver.gecko.driver", "/home/ato/bin");
+```
+
+Or by passing it as a flag to the [java(1)](http://www.manpagez.com/man/1/java/) launcher:
+
+	% java -Dwebdriver.gecko.driver=/home/ato/bin YourApplication
+
+Your milage with this approach may vary
+based on which programming language bindings you are using.
+It is in any case generally the case that geckodriver will be picked up
+if it is available on the system path.
+In a bash compatible shell,
+you can make other programs aware of its location
+by exporting or setting the `PATH` variable:
+
+	% export PATH=$PATH:/home/ato/bin
+	% whereis geckodriver
+	geckodriver: /home/ato/bin/geckodriver
+
+On Window systems you can change the system path
+by right-clicking **My Computer** and choosing **Properties**.
+The dialogue that appears, navigate
+**Advanced** → **Environmental Variables** → **Path**.
+
+Or in the Windows console window:
+
+	$ set PATH=%PATH%;C:\bin\geckodriver
+
+### Standalone
+
+Since geckodriver is a separate HTTP server
+that is a complete remote end implementation of [WebDriver](https://w3c.github.io/webdriver/webdriver-spec.html),
+it is possible to circumvent the Selenium remote server
```

Comment:
I might say "it is possible to avoid using the Selenium remote server if you have no requirements to[…]"